Common/SecurityIdentifier.cs

```diff
@@ -629,12 +632,20 @@ private static bool TryParse(string value, out SecurityIdentifier identifier, ou
         private static bool TryParseProperties(string value, out Exception exception, out SecurityIdentifier identifier)
         {
             exception = null;
-            identifier = Empty;
 
             if (string.IsNullOrWhiteSpace(value))
+            {
+                identifier = Empty;
+                return true;
+            }
+
+            // for performance, we first verify if we already have parsed this SecurityIdentifier
+            if (SecurityIdentifierCache.TryGetValue(value, out identifier))
```

Comment:
this cache will forever grow, although I'm not sure if it would pose a problem, and any attempt to make the cache smart would only eat up the performance gain -- TLDR -- I think this is good as-is.

Reply:
True 👍  this improvement has a big impact on fundamental data source where for each day we reparse the same value into a new `SecurityIdentifier`